python/llm/example/transformers_int4.py

```diff
@@ -0,0 +1,38 @@
+#
+# Copyright 2016 The BigDL Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import torch
+from transformers import LlamaTokenizer, LlamaForCausalLM
+
+if __name__ == '__main__':
+    model_path = 'decapoda-research/llama-7b-hf'
+    model = LlamaForCausalLM.from_pretrained(model_path, torch_dtype=torch.float32)
+
+    ###### Quantize model into ggml q4_0 format ######
+    from bigdl.llm.ggml.transformers import ggml_convert_int4
+    model = ggml_convert_int4(model)
+    print(model)
+    ###################################################
```

Comment:
Change to:
```python
from bigdl.llm.transformers import AutoModelForCausalLM
model = AutoModelForCausalLM.from_pretrained("model_path", ..., load_in_4bit=True)
```

Reply:
> Change to:
> 
> ```python
> from bigdl.llm.transformers import AutoModelForCausalLM
> model = AutoModelForCausalLM.from_pretrained("model_path", ..., load_in_4bit=True)
> ```

We also have a `bigdl.llm.ggml.transformers.AutoModelForCausalLM`, should we merge these two classes?

Reply:
Yes. I think it support three cases today:
1) load bigdl-llm converted INT4 models
2) load huggingface model and convert using bigdl-llm convert_model
3) load huggingface model and convert linear layers to INT4

We need to define the right APIs for different cases.